iFixit/react-commerce
# Monorepo setup

This PR closes #119  

The purpose of this PR is to have Strapi config under version control, so that we can easily create multiple environments as well as support local development. I've discussed with @djmetzle how to setup the staging environments, and although ideally we would like to have as many staging environments as PRs that are open, we are taking a simpler approach for now. Once this PR is merged @djmetzle will setup a couple of staging environments, both based on the `main` branch.

## Changelog

- Moves root into `/frontend` folder
- Adds `/backend` folder with Strapi config files
- If `SEED_DB` env variable is `true` it seeds the database (useful to quickly bootstrap local development)
- Adds docker-compose file to boostrap Strapi server along with a Postgres db. This is used during CI to run integration tests now and can be used to deploy staging servers
- During CI tests run against a local instance of Strapi

## QA

Until we setup the staging servers we need to clone the repo and test it locally.

1. Clone this repo
2. Make sure you are running on node v14 (if you are using nvm run `nvm use` in the project working directory)
3. Install dependencies `npm install`
4. Rename `backend/.env.example` to `backend/.env` (this will enable db seed)
5. Run dev server: `npm run dev`
6. Make sure that everything compiles and the website runs correctly

### Access to local strapi instance

Email: `strapi@ifixit.com`
Password: `Password1`
